### PR TITLE
feat(build_container): help musl-tools find headers

### PR DESCRIPTION
**Pull request by ShadowCurse**
### Summary of the PR

This PR is an addition to the #97 PR and helps musl-tools to find linux headers.

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [ ] All commits in this PR are signed (with `git commit -s`), and the commit
  message has max 60 characters for the summary and max 75 characters for each
  description line.
- [ ] All added/changed functionality has a corresponding unit/integration
  test.
- [ ] All added/changed public-facing functionality has entries in the "Upcoming 
  Release" section of CHANGELOG.md (if no such section exists, please create one).
- [ ] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

**Review by JonathanWoollett-Light — APPROVED** [review]

**Review by roypat — APPROVED** [review]

**Comment by stefano-garzarella** [comment]
@ShadowCurse it looks like there is some issue with `ARCH` env:

```
#12 35.43 + cd /usr/include/-linux-musl
#12 35.43 /opt/src/scripts/build_container.sh: line 21: cd: /usr/include/-linux-musl: No such file or directory
```

https://github.com/rust-vmm/rust-vmm-container/actions/runs/7400336298/job/20133743257#step:8:1531

**Comment by ShadowCurse** [comment]
Hi @stefano-garzarella, thx for noticing this issue. I will see what went wrong.

File: build_container.sh [hunk 1]
```diff
@@ -17,6 +17,12 @@ DEBIAN_FRONTEND="noninteractive" apt-get install --no-install-recommends -y \
 # cleanup
 apt-get clean && rm -rf /var/lib/apt/lists/*
 
+# help musl-gcc find linux headers
```
Comment (stefano-garzarella) [inline build_container.sh:20]: I'd like to know why we need this workaround, is it something specific for this image (Ubuntu:22.04)?

Comment (ShadowCurse) [reply build_container.sh:20]: Without it compiling with musl-gcc fails like this: https://buildkite.com/rust-vmm/vhost-ci/builds/615#018cd414-a613-4e9f-804f-c698d683cf64
From what I understand, when compiling, `musl-gcc` will only look at the `musl` specific headers (in `/usr/include/$ARCH-linux-musl`) and will not find default linux headers. So we need to manually link linux headers to the `musl` directory.

Comment (stefano-garzarella) [reply build_container.sh:20]: Yes, but my question is why doesn't the distro do that already?
Is it a bug? Or do they expect us to have to configure `musl-gcc` differently?

Also, are we sure libraries/executables compiled this way with `musl` will work?

Comment (ShadowCurse) [reply build_container.sh:20]: > Yes, but my question is why doesn't the distro do that already?

I would like to know the answer to this question myself.

> Also, are we sure libraries/executables compiled this way with musl will work?

Considering we just point the compiler to the system headers, I don't see why would final lib/exec will not work. Also this technique is used in FC: https://github.com/firecracker-microvm/firecracker/blob/ab3e4fe129225c9620bd6591ea8ab3d0f03337f0/tools/devctr/Dockerfile#L141 and it never caused any problems.

Comment (stefano-garzarella) [reply build_container.sh:20]: > > Yes, but my question is why doesn't the distro do that already?
> 
> I would like to know the answer to this question myself.

eheh :-)

> 
> > Also, are we sure libraries/executables compiled this way with musl will work?
> 
> Considering we just point the compiler to the system headers, I don't see why would final lib/exec will not work. Also this technique is used in FC: https://github.com/firecracker-microvm/firecracker/blob/ab3e4fe129225c9620bd6591ea8ab3d0f03337f0/tools/devctr/Dockerfile#L141 and it never caused any problems.

Okay, I was just looking for confirmation on that. Next time I suggest explaining better (at least in the commit description), how it was tested or inspired, so it's easier for reviewers.

**Pull request merged**